tests/aggregation/tests.py

```diff
@@ -395,6 +395,57 @@ def test_count(self):
         vals = Book.objects.aggregate(Count("rating", distinct=True))
         self.assertEqual(vals, {"rating__count": 4})
 
+    def test_nonselected_annotations_not_in_groupby(self):
+        vals = list(Book.objects.annotate(xprice=F('price'))
+                                .filter(xprice__lte=30)
+                                .values('publisher')
+                                .annotate(count=Count('pk'))
+                                .order_by('publisher'))
+        self.assertEqual(
+            vals, [
+                {
+                    'publisher': 1,
+                    'count': 2,
+                },
+                {
+                    'publisher': 2,
+                    'count': 1,
+                },
+                {
+                    'publisher': 3,
+                    'count': 1,
+                },
+            ]
+        )
+
+    def test_nonselected_annotations_not_in_groupby2(self):
+        vals = list(Book.objects.annotate(xprice=F('price'))
```

Comment:
A style that didn't require so many lines and use non-4-space indent would be more consistent with the rest of the code. Something like:
``` 
vals = list(Book.objects.annotate(xprice=F('price')).filter(xprice__lte=30)
    .values('publisher', 'contact').annotate(count=Count('pk')).values('publisher', 'count')
    .order_by('publisher'))
```